plonky3/src/params/mersenne_31.rs

```diff
@@ -5,27 +5,24 @@ use std::marker::PhantomData;
 
 use lazy_static::lazy_static;
 
-use crate::params::{Challenger, FieldElementMap, Plonky3Field};
+use crate::params::{
+    poseidon2::{poseidon2_external_constants, poseidon2_internal_constants},
+    Challenger, FieldElementMap, Plonky3Field,
+};
 use p3_challenger::DuplexChallenger;
 use p3_circle::CirclePcs;
 use p3_commit::ExtensionMmcs;
 use p3_field::{extension::BinomialExtensionField, Field};
 use p3_fri::FriConfig;
 use p3_merkle_tree::MerkleTreeMmcs;
 use p3_mersenne_31::{DiffusionMatrixMersenne31, Mersenne31};
-use p3_poseidon2::{Poseidon2, Poseidon2ExternalMatrixGeneral};
+use p3_poseidon2::{poseidon2_round_numbers_128, Poseidon2, Poseidon2ExternalMatrixGeneral};
 use p3_symmetric::{PaddingFreeSponge, TruncatedPermutation};
 use p3_uni_stark::StarkConfig;
 
-use rand::{distributions::Standard, Rng, SeedableRng};
-
 use powdr_number::Mersenne31Field;
 
 const D: u64 = 5;
-// params directly taken from plonky3's poseidon2_round_numbers_128 function
-// to guarantee 128-bit security.
-const ROUNDS_F: usize = 8;
-const ROUNDS_P: usize = 12;
```

Comment:
Looks like these where actually wrong! See [this line](https://github.com/Plonky3/Plonky3/blob/aaa9b07690a6dc847a800553ada490c53600981e/poseidon2/src/round_numbers.rs#L41).